src/emmetHelper.ts

```diff
@@ -123,12 +123,13 @@ export function doComplete(document: TextDocument, position: Position, syntax: s
 		const stylesheetCustomSnippetsKeys = stylesheetCustomSnippetsKeyCache.has(syntax) ? stylesheetCustomSnippetsKeyCache.get(syntax) : stylesheetCustomSnippetsKeyCache.get('css');
 		completionItems = makeSnippetSuggestion(stylesheetCustomSnippetsKeys, currentWord, abbreviation, abbreviationRange, expandOptions, 'Emmet Custom Snippet', false);
 
-		if (!completionItems.find(x => x.textEdit.newText === expandedAbbr.textEdit.newText)) {
-
+		if (!completionItems.find(x => x.textEdit.newText === expandedAbbr.textEdit.newText)) {	
+			
 			// Fix for https://github.com/Microsoft/vscode/issues/28933#issuecomment-309236902
 			// When user types in propertyname, emmet uses it to match with snippet names, resulting in width -> widows or font-family -> font: family
 			// Filter out those cases here.
-			const abbrRegex = new RegExp('.*' + abbreviation.split('').map(x => x === '$' ? '\\$' : x).join('.*') + '.*', 'i');
+			let abbreviationWithoutDashPrefix = abbreviation[0] == '-' ? abbreviation.slice(1) : abbreviation;
```

Comment:
This is only applicable for css abbreviations. So I'd suggest to have a another function `vendorPrefixCssAbbreviation` that takes in abbreviation and syntax and returns a boolean. You can use `isStylesheet` to check the syntax as well as look for the preceeding `-`.

Re-use this function inside `expandAbbreviationWithDash`
- rename this function as `expandCssAbreviationWithVendorPrefixes`
- have a check on the syntax. Use the `isStylesheet` function.

Also rename `expandAbbreviationWithDash` to `expandCssAbbreviationWithVendorPrefix`